### PR TITLE
WIP add StringTokenazer_woekOnBugs

### PR DESCRIPTION
**Pull request by PavelBoPavel**

### CODE REVIEW

**Review by KaydunovDenis — CHANGES_REQUESTED** [review]

File: homework/PavelBo/collections/stringTokenizer/src/main/java/com/tutrit/java/quickstart/StringAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    public List<String> splitToWords(String text){
+        return Collections.list(new StringTokenizer(text, ",. ")).stream()
+                .map(token -> (String) token)
+                .collect(Collectors.toList());
+    }
+
+    public List<String> splitToPunctuationMarks(String text){
+        return Collections.list(new StringTokenizer(text, " йцукенгшщзхъфывапролджэячсмитьбюЙЦУКЕНГШЩЗХЪФЫВАПРОЛДЖЭЯЧСМИТЬБЮ")).stream()
```
[the remark this answers is not in the document]
Comment (PavelBoPavel) [reply homework/PavelBo/collections/stringTokenizer/src/main/java/com/tutrit/java/quickstart/StringAnalyzer.java:26]: в StringTokenazer регулярных выражений таких как в методе split() не нашел.

File: homework/PavelBo/collections/stringTokenizer/src/main/java/com/tutrit/java/quickstart/StringAnalyzer.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    private void listToMap(List<String> punctuationMarksList, Map<String, Integer> punctuationMarksMap) {
+        for (String punctuationMarks : punctuationMarksList){
+            if (punctuationMarksMap.containsKey(punctuationMarks)){
+                punctuationMarksMap.entrySet().stream().forEach(punctuationMark -> {
+                    if (punctuationMark.getKey().equals(punctuationMarks)) {
+                        int value = punctuationMark.getValue() + 1;
+                        punctuationMark.setValue(value);
+                    }
+                });
```
Comment (max-java) [inline homework/PavelBo/collections/stringTokenizer/src/main/java/com/tutrit/java/quickstart/StringAnalyzer.java:63]: usually we don't mix streams with foreach 
also 
``` 
for {
    if {
        if {
        }
    }
}
``` 
is too complicated. I suggest you revise this and split into separate methods.

File: homework/PavelBo/collections/stringTokenizer/src/test/java/com/tutrit/java/quickstart/StringAnalyzerTest.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+
+import static junit.framework.Assert.assertEquals;
+
+
+public class StringAnalyzerTest{
```
Comment (max-java) [inline homework/PavelBo/collections/stringTokenizer/src/test/java/com/tutrit/java/quickstart/StringAnalyzerTest.java:11]: Nevertheless, it woks!

**Review by max-java — CHANGES_REQUESTED** [review]
ptal and let''s revise it once we reach the streams topic

**Pull request closed without merging**